### PR TITLE
Wireguard Debian install and BATS 

### LINKED ISSUE #1027: Wireguard BATS tests
@mstormi This is a log of what happens, the problem appears to be that you have targeted this script to only run on a RPi
 and Docker / Travis does not like that.
See the changes I made to the BATS script to get the proper output in #1026 
```
2020-07-10_16:46:11_UTC [openHABian] Installing Wireguard and enabling VPN remote access... Executing: /tmp/apt-key-gpghome.9ZZrE3a30k/gpg.1.sh --keyserver keyserver.ubuntu.com --recv-keys 04EE7237B7D453EC
gpg: key E0B11894F66AEC98: "Debian Archive Automatic Signing Key (9/stretch) <ftpmaster@debian.org>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
Executing: /tmp/apt-key-gpghome.NnWaUtGVM3/gpg.1.sh --keyserver keyserver.ubuntu.com --recv-keys 648ACFD622F3D138
gpg: key DC30D7C23CBBABEE: "Debian Archive Automatic Signing Key (10/buster) <ftpmaster@debian.org>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1

$ apt-get update 
Get:1 http://security.debian.org/debian-security buster/updates InRelease [65.4 kB]
Get:2 http://deb.debian.org/debian buster InRelease [121 kB]
Get:3 http://deb.debian.org/debian buster-updates InRelease [51.9 kB]
Get:4 http://security.debian.org/debian-security buster/updates/main amd64 Packages [208 kB]
Get:5 http://deb.debian.org/debian unstable InRelease [146 kB]
Get:6 http://deb.debian.org/debian buster/main amd64 Packages [7905 kB]
Get:7 http://deb.debian.org/debian buster-updates/main amd64 Packages [7868 B]
Get:8 http://deb.debian.org/debian unstable/main amd64 Packages [8272 kB]
Fetched 16.8 MB in 11s (1571 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package raspberrypi-kernel-headers
dpkg-query: package 'wireguard-dkms' is not installed and no information is available
Use dpkg --info (= dpkg-deb --info) to examine archive files.
/usr/sbin/dpkg-reconfigure: wireguard-dkms is not installed
/opt/openhabian/./functions/vpn.bash: line 62: modprobe: command not found
/opt/openhabian/./functions/vpn.bash: line 65: /etc/wireguard/server_public_key: No such file or directory
tee: /opt/openhabian/./functions/vpn.bash: line 65: wg: command not found
/etc/wireguard/server_private_key: No such file or directory
/opt/openhabian/./functions/vpn.bash: line 66: /etc/wireguard/client_public_key: No such file or directory
/opt/openhabian/./functions/vpn.bash: line 66: wg: command not found
tee: /etc/wireguard/client_private_key: No such file or directory
chown: cannot access '/etc/wireguard': No such file or directory
ERROR:systemctl:Unit wg-quick@wg0.service could not be found.
OK
# 2020-07-10_16:46:28_UTC [openHABian] Wireguard VPN installation successful.
# 2020-07-10_16:46:28_UTC [openHABian] Wireguard VPN setup starting...
2020-07-10_16:46:28_UTC [openHABian] Installing Wireguard required packages (dnsutils)... 
$ apt-get install --yes dnsutils 
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  bind9-host libbind9-161 libdns1104 libfstrm0 libgeoip1 libirs161 libisc1100
  libisccc161 libisccfg163 liblmdb0 liblwres161 libprotobuf-c1
Suggested packages:
  rblcheck geoip-bin
Recommended packages:
  geoip-database
The following NEW packages will be installed:
  bind9-host dnsutils libbind9-161 libdns1104 libfstrm0 libgeoip1 libirs161
  libisc1100 libisccc161 libisccfg163 liblmdb0 liblwres161 libprotobuf-c1
0 upgraded, 13 newly installed, 0 to remove and 20 not upgraded.
Need to get 3738 kB of archives.
After this operation, 9676 kB of additional disk space will be used.
Get:1 http://security.debian.org/debian-security buster/updates/main amd64 libisc1100 amd64 1:9.11.5.P4+dfsg-5.1+deb10u1 [457 kB]
Get:2 http://deb.debian.org/debian buster/main amd64 libfstrm0 amd64 0.4.0-1 [20.8 kB]
Get:3 http://deb.debian.org/debian buster/main amd64 libgeoip1 amd64 1.6.12-1 [93.1 kB]
Get:4 http://security.debian.org/debian-security buster/updates/main amd64 libdns1104 amd64 1:9.11.5.P4+dfsg-5.1+deb10u1 [1222 kB]
Get:5 http://deb.debian.org/debian buster/main amd64 liblmdb0 amd64 0.9.22-1 [45.0 kB]
Get:6 http://deb.debian.org/debian buster/main amd64 libprotobuf-c1 amd64 1.3.1-1+b1 [26.5 kB]
Get:7 http://security.debian.org/debian-security buster/updates/main amd64 libisccc161 amd64 1:9.11.5.P4+dfsg-5.1+deb10u1 [236 kB]
Get:8 http://security.debian.org/debian-security buster/updates/main amd64 libisccfg163 amd64 1:9.11.5.P4+dfsg-5.1+deb10u1 [266 kB]
Get:9 http://security.debian.org/debian-security buster/updates/main amd64 libbind9-161 amd64 1:9.11.5.P4+dfsg-5.1+deb10u1 [247 kB]
Get:10 http://security.debian.org/debian-security buster/updates/main amd64 liblwres161 amd64 1:9.11.5.P4+dfsg-5.1+deb10u1 [251 kB]
Get:11 http://security.debian.org/debian-security buster/updates/main amd64 bind9-host amd64 1:9.11.5.P4+dfsg-5.1+deb10u1 [271 kB]
Get:12 http://security.debian.org/debian-security buster/updates/main amd64 libirs161 amd64 1:9.11.5.P4+dfsg-5.1+deb10u1 [237 kB]
Get:13 http://security.debian.org/debian-security buster/updates/main amd64 dnsutils amd64 1:9.11.5.P4+dfsg-5.1+deb10u1 [365 kB]
Fetched 3738 kB in 3s (1336 kB/s)
Selecting previously unselected package libfstrm0:amd64.
(Reading database ... 25468 files and directories currently installed.)
Preparing to unpack .../00-libfstrm0_0.4.0-1_amd64.deb ...
Unpacking libfstrm0:amd64 (0.4.0-1) ...
Selecting previously unselected package libgeoip1:amd64.
Preparing to unpack .../01-libgeoip1_1.6.12-1_amd64.deb ...
Unpacking libgeoip1:amd64 (1.6.12-1) ...
Selecting previously unselected package liblmdb0:amd64.
Preparing to unpack .../02-liblmdb0_0.9.22-1_amd64.deb ...
Unpacking liblmdb0:amd64 (0.9.22-1) ...
Selecting previously unselected package libprotobuf-c1:amd64.
Preparing to unpack .../03-libprotobuf-c1_1.3.1-1+b1_amd64.deb ...
Unpacking libprotobuf-c1:amd64 (1.3.1-1+b1) ...
Selecting previously unselected package libisc1100:amd64.
Preparing to unpack .../04-libisc1100_1%3a9.11.5.P4+dfsg-5.1+deb10u1_amd64.deb ...
Unpacking libisc1100:amd64 (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Selecting previously unselected package libdns1104:amd64.
Preparing to unpack .../05-libdns1104_1%3a9.11.5.P4+dfsg-5.1+deb10u1_amd64.deb ...
Unpacking libdns1104:amd64 (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Selecting previously unselected package libisccc161:amd64.
Preparing to unpack .../06-libisccc161_1%3a9.11.5.P4+dfsg-5.1+deb10u1_amd64.deb ...
Unpacking libisccc161:amd64 (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Selecting previously unselected package libisccfg163:amd64.
Preparing to unpack .../07-libisccfg163_1%3a9.11.5.P4+dfsg-5.1+deb10u1_amd64.deb ...
Unpacking libisccfg163:amd64 (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Selecting previously unselected package libbind9-161:amd64.
Preparing to unpack .../08-libbind9-161_1%3a9.11.5.P4+dfsg-5.1+deb10u1_amd64.deb ...
Unpacking libbind9-161:amd64 (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Selecting previously unselected package liblwres161:amd64.
Preparing to unpack .../09-liblwres161_1%3a9.11.5.P4+dfsg-5.1+deb10u1_amd64.deb ...
Unpacking liblwres161:amd64 (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Selecting previously unselected package bind9-host.
Preparing to unpack .../10-bind9-host_1%3a9.11.5.P4+dfsg-5.1+deb10u1_amd64.deb ...
Unpacking bind9-host (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Selecting previously unselected package libirs161:amd64.
Preparing to unpack .../11-libirs161_1%3a9.11.5.P4+dfsg-5.1+deb10u1_amd64.deb ...
Unpacking libirs161:amd64 (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Selecting previously unselected package dnsutils.
Preparing to unpack .../12-dnsutils_1%3a9.11.5.P4+dfsg-5.1+deb10u1_amd64.deb ...
Unpacking dnsutils (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Setting up liblmdb0:amd64 (0.9.22-1) ...
Setting up libfstrm0:amd64 (0.4.0-1) ...
Setting up libprotobuf-c1:amd64 (1.3.1-1+b1) ...
Setting up libgeoip1:amd64 (1.6.12-1) ...
Setting up libisc1100:amd64 (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Setting up liblwres161:amd64 (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Setting up libisccc161:amd64 (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Setting up libdns1104:amd64 (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Setting up libisccfg163:amd64 (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Setting up libbind9-161:amd64 (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Setting up libirs161:amd64 (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Setting up bind9-host (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Setting up dnsutils (1:9.11.5.P4+dfsg-5.1+deb10u1) ...
Processing triggers for libc-bin (2.28-10) ...
OK
cat: /etc/wireguard/server_private_key: No such file or directory
cat: /etc/wireguard/server_public_key: No such file or directory
cat: /etc/wireguard/client_private_key: No such file or directory
cat: /etc/wireguard/client_public_key: No such file or directory
/opt/openhabian/./functions/vpn.bash: line 121: /etc/wireguard/wg0.conf: No such file or directory
/opt/openhabian/./functions/vpn.bash: line 122: /etc/wireguard/wg0-client.conf: No such file or directory
chmod: cannot access '/etc/wireguard/*': No such file or directory
2020-07-10_16:46:36_UTC [openHABian] Generating QR to load config on the client side (download Wireguard app from PlayStore or AppStore)... /opt/openhabian/./functions/vpn.bash: line 162: /etc/wireguard/wg0-client.conf: No such file or directory
not ok 1 destructive-wireguard_install
# (in test file ./functions/vpn.bats, line 21)
#   `[ "$status" -eq 0 ]' failed
```

I don't really know much about the install script so I can't help to much with a fix but hopefully that gets you on the right track.

### PR DESCRIPTION
**Pull request by mstormi**
Splits BATS tests into java-* to run on 64bit VM and rest to run on 32bit.

Fixes #1027 

Signed-off-by: Markus Storm markus.storm@gmx.net

### CODE REVIEW

File: ci-setup.bash [hunk 1]
```diff
@@ -16,8 +16,8 @@ echo_process() {
 
 if [[ $1 == "travis" ]]; then
   # prepare configuration for tests, select debug level here:
-  sed -i 's#debugmode=.*$#debugmode=on#' build-image/openhabian.conf
-  #sed -i 's#debugmode=.*$#debugmode=maximum#' build-image/openhabian.conf
+  #sed -i 's#debugmode=.*$#debugmode=on#' build-image/openhabian.conf
+  sed -i 's#debugmode=.*$#debugmode=maximum#' build-image/openhabian.conf
```
Comment (ecdye) [inline ci-setup.bash:20]: Be sure to change back before merge

File: functions/vpn.bash [hunk 2]
```diff
@@ -43,26 +43,31 @@ install_wireguard() {
...
+      if is_debian; then
+        echo 'deb http://deb.debian.org/debian buster-backports main contrib non-free' > /etc/apt/sources.list.d/wireguard.list
+      else
+        echo "FAILED (unsupported OS)"; return 1
+      fi
     fi
+    if ! cond_redirect apt-get update &>/dev/null; then echo "FAILED (update apt lists)"; return 1; fi
   fi
-  apt-get install --yes wireguard wireguard-dmks wireguard-tools qrencode
+  cond_redirect apt-get install --yes wireguard qrencode &>/dev/null
```
Comment (ecdye) [inline functions/vpn.bash:66]: Why all of the output redirecting?

File: functions/vpn.bats [hunk 3]
```diff
@@ -3,29 +3,34 @@
...
+  echo "$output" >&3
   [ "$status" -eq 0 ]
   echo -e "# ${COL_GREEN}$(timestamp) [openHABian] Wireguard VPN setup successful.${COL_DEF}" >&3
 
-  echo -e "# ${COL_CYAN}$(timestamp) [openHABian] Checking if Wireguard service is running...${COL_DEF}" >&3
-  run systemctl is-active wg-quick@wg0.service 3>&-
-  if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
-  [ "$status" -eq 0 ]
-  echo -e "# ${COL_GREEN}$(timestamp) [openHABian] Wireguard service is running.${COL_DEF}" >&3
+  # that will not work inside Docker ...
```
Comment (ecdye) [inline functions/vpn.bats:28]: Why not?

File: functions/vpn.bash (hunk 2 above)
Comment (mstormi) [reply functions/vpn.bash]: Right, I've cut that back. It should be there if there's problems, i.e. when BATS $status != 0

File: functions/vpn.bats (hunk 3 above)
Comment (mstormi) [reply functions/vpn.bats:32]: Loading kernel modules does not seem to work in Docker hence the service doesn't run.
BTW changing kernel parameters doesn't either (it's refused as "read-only").

Comment (ecdye) [reply functions/vpn.bats:27]: Hmm, nothing is ever perfect, oh well guess we will need to make do with what works.

**Pull request merged**